sparse_ho/ho.py

```diff
@@ -6,41 +6,47 @@
 
 
 def grad_search(
-        algo, criterion, model, X, y, log_alpha0, monitor, n_outer=100, verbose=False,
-        tolerance_decrease='constant', tol=1e-5, t_max=10000):
+        algo, criterion, model, X, y, log_alpha0, monitor, n_outer=100,
+        verbose=False, tolerance_decrease='constant', tol=1e-5, t_max=10000):
     """This line-search code is taken from here:
     https://github.com/fabianp/hoag/blob/master/hoag/hoag.py
 
     Parameters
     ----------
-    algo: TODO
-    criterion: TODO
-    model: TODO
-    log_alpha0: float
-        log of the regularization coefficient alpha
-    tol : float
-        tolerance for the inner optimization solver
-    monitor: Monitor object
-        used to store the value of the cross-validation function
-    n_outer: int
-        number of maximum iteration in the outer loop (for the line search)
-    tolerance_decrease: string
-        tolerance decrease strategy for approximate gradient
-    criterion: string
+    algo: instance of BaseAlgo
```

Comment:
guys these classes do not actually exist !

see my attempt to solve this : https://github.com/QB3/sparse-ho/pull/52

this still needs to be fixed with a BaseAlgo and your API for criterion is not consistent as revealed by the failing tests.